keps/sig-network/2943-networkpolicy-status/README.md

```diff
@@ -0,0 +1,641 @@
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Design Details](#design-details)
+  - [Test Plan](#test-plan)
+  - [Graduation Criteria](#graduation-criteria)
+  - [Upgrade / Downgrade Strategy](#upgrade--downgrade-strategy)
+  - [Version Skew Strategy](#version-skew-strategy)
+- [Production Readiness Review Questionnaire](#production-readiness-review-questionnaire)
+  - [Feature Enablement and Rollback](#feature-enablement-and-rollback)
+  - [Rollout, Upgrade and Rollback Planning](#rollout-upgrade-and-rollback-planning)
+  - [Monitoring Requirements](#monitoring-requirements)
+  - [Dependencies](#dependencies)
+  - [Scalability](#scalability)
+  - [Troubleshooting](#troubleshooting)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Infrastructure Needed (Optional)](#infrastructure-needed-optional)
+<!-- /toc -->
+
+## Release Signoff Checklist
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [ ] (R) Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
+- [ ] (R) KEP approvers have approved the KEP status as `implementable`
+- [ ] (R) Design details are appropriately documented
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input (including test refactors)
+  - [ ] e2e Tests for all Beta API Operations (endpoints)
+  - [ ] (R) Ensure GA e2e tests for meet requirements for [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [ ] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+
+NetworkPolicy objects does not contain a status subresource. This KEP 
+proposes to add this subresource in NetworkPolicy objects, allowing CNIs to provide 
+feedback to users whether a NetworkPolicy and its features has been properly parsed.
+
+## Motivation
+
+While implementing new features in Network Policy, we've realized that some CNIs were 
```

Comment:
It sounds like these scenarios are for providers that only partially implement the network policy API, or have some restrictions / limitations in the objects they will accept from the API.

If that is the case, should we be handling this via some form of validation so that users get immediate feedback rather than needing to seek it out?

If I create an "invalid" policy for my cluster, I'd like to know about it at the time I attempt to create it, rather than a few weeks down the road when something unexpected happens.

Reply:
I agree with you Casey, but in this case that should be some common Validation Webhook right?

Also, just reminding that this Status field was mostly a PRR request to provide some 'feature usage' feedback

Reply:
Yeah, I didn't mean to imply a solution. I was just suggesting that these use-cases sound better suited to validation as far as user-experience goes.

Could either be a webhook, some sort of capability negotiation, etc. Haven't thought that far into it!

Reply:
webhook would be the responsibility of individual network policy providers and not something that will be part of k/k.. right? but i agree that would be most appropriate, but not something we can solve with a KEP

Reply:
> webhook would be the responsibility of individual network policy providers 

Yes, but populating the conditions field of a NetworkPolicy status is also ultimately up to the network policy provider to actually implement.